engine/src/main/java/org/hibernate/search/analyzer/spi/AnalyzerDefiner.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.analyzer.spi;
```

Comment:
I would not make that an public SPI for now, esp. as the solution might change again if we can do the proper validation of the analyzer. Or at least marking it as experimental.